pkg/framework/resource/resource.go

```diff
@@ -96,6 +96,23 @@ func (r *Resource) InstallResource(name, values, channel string, conditions ...f
 	return nil
 }
 
+func (r *Resource) UpdateResource(name, values, channel string, conditions ...func() error) error {
+	chartValuesEnv := os.ExpandEnv(values)
```

Comment:
I am totally against all kinds of env magic here. Again. This is a library. In its depth no env magic should take place. The primitives should work against deterministic inputs.

Reply:
I agree, the reason I left this here is that most of our `values` we use currently are expanded through env. I am fine with removing it, but then we will need to have a more complex transition when we want to use these functions in most of our e2e tests.

Reply:
Why? When it is necessary to still use the env then you can just do it before calling the lib. Then it is visible in the tests.